caddyhttp/httpserver/server.go

```diff
@@ -333,6 +334,12 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Server", "Caddy")
 
+	if requestid.IsActive {
+		reqid := requestid.UUID()
+		c := context.WithValue(r.Context(), caddy.CtxKey("REQUESTID"), reqid)
+		r = r.WithContext(c)
+	}
```

Comment:
- Move this into an HTTP handler like every other middleware should be; you can then lose the whole `IsActive` variable.
- I generally try to avoid storing stuff in the context if possible, but I think a request ID is one of the few things that would be expected to belong in a context. So that is OK. However...
- ... There's a little trick you can use to set a value on a replacer. The proxy middleware, for example, defines an `{upstream}` placeholder like this:

```
if rr, ok := w.(*httpserver.ResponseRecorder); ok && rr.Replacer != nil {
	rr.Replacer.Set("upstream", host.Name)
}
```

You probably should do this for request ID, and then you can also add it to the context as well.